pulseapi/profiles/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+
+class UserProfile(models.Model):
+    """
+    This class houses all user profile information,
+    such as real name, social media information,
+    bookmarks on the site, etc.
+    """
+
+    user = models.ForeignKey(
+        'users.EmailUser',
+        on_delete=models.SET_NULL,
+        related_name='profile',
+        null=True
+    )
+
+    # "user X bookmarked entry Y" is a many to many relation,
+    # for which we also want to know *when* a user bookmarked
+    # a specific entry. As such, we use a helper class that
+    # tracks this relation as well as the time it's created.
+    bookmarks = models.ManyToManyField(
+        'entries.Entry',
+        through='profiles.UserBookmarks'
+    )
+
+    def __str__(self):
+        return 'profile for {}'.format(self.user.email)
+
+
+class UserBookmarks(models.Model):
```

Comment:
nit: probably need a `Meta` class to specify the verbose name as `Bookmark` so it doesn't add the extra `s` - `Bookmarkss`